File: Android/iOrder/app/src/main/java/hr/fer/oobl/iorder/iorder/base/BaseActivity.java
package hr.fer.oobl.iorder.iorder.base;

import hr.fer.oobl.iorder.iorder.injection.activity.DaggerActivity;

public abstract class BaseActivity extends DaggerActivity implements BaseView {

    @Override
    public void onStop() {
        getPresenter().stop();
        super.onStop();
    }

    @Override
    public void onPause() {
        getPresenter().deactivate();
        super.onPause();
    }

    public abstract ScopedPresenter getPresenter();
}

File: Android/iOrder/app/src/main/java/hr/fer/oobl/iorder/iorder/base/BaseActivity.kt
package hr.fer.oobl.iorder.iorder.base


import hr.fer.oobl.iorder.iorder.injection.activity.DaggerActivity

abstract class BaseActivity : DaggerActivity(), BaseView {

    abstract val presenter: ScopedPresenter

    public override fun onStop() {
        presenter.stop()
        super.onStop()
    }

    public override fun onPause() {
        presenter.deactivate()
        super.onPause()
    }
}
